tdms/src/hdf5_io/hdf5_base.cpp

```diff
@@ -49,8 +50,58 @@ H5Dimension HDF5Base::shape_of(const std::string &group_name,
 }
 
 bool HDF5Base::is_ok() const {
-  // TODO: check for file health might be unnessicary given we've constructed
+  // TODO: check for file health might be unnecessary given we've constructed
   // the object.
   return file_->isHdf5(filename_);
   // return file_->isAccessible(filename_) && file_->isHdf5(filename_);
 }
+
+bool HDF5Base::flagged_MATLAB_empty(const std::string &object_path) const {
+  // Can't check anything if there's no file
+  if (file_ == nullptr) { throw std::runtime_error("No file opened"); }
```

Comment:
I'd use our internal healthcheck:
```suggestion
  if (!is_ok()) { throw std::runtime_error("Problem with the file!"); }
```
And put the `file_ != nullptr` check up in `HDF5Base::is_ok`.

Although by opening the `H5::File` in the constructor, we're guaranteeing that `file_` can never be null (and probably can never _not_ be `isHdf5`).

Reply:
If we delete explicitly the default constructor, I think we're safe.

That is: it never makes sense to have a `HDF5Base` without a `file_` which needs a `path` and `mode` which means our constructor _must_ provide these.

... what do you think? (Not for this PR)

Reply:
Sounds sensible - the default constructor is already `protected` anyway so we can't actually use it to instantiate `HDF5Base` nor any of the derived classes.